### PR TITLE
Show warnings when loading a mismatched or outdated mission

### PR DESCRIPTION
**Pull request by jaj22**
## What type of PR is this.
1. [ ] Bug
2. [X] Change
3. [ ] Enhancement

### What have you changed and why?
Running a pre-mod Antistasi mission together with the Antistasi mod will break. Users need to be informed that they're running the wrong mission or an incompatible combination.

Decided to put the warning code in preInit so that it also warns for Barbolani missions and others without the initServer/initClient proxies. I've commented out the code in prepFunctions that blocks running preInit functions with a mismatched mission so that this works. This also makes the behaviour consistent in debug vs normal.

Missions that do use the initServer proxy also bail out in initServer. This avoids hiding the warning behind the setup interface and prevents potential damage.

Also added similar functionality to check mod/extension mission version against a requirement. This is unlikely to be used for a while but it's important to get the mission version config in now.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

Should test DS, but no reason that it shouldn't work given the methods.

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
- Load a mod mission.
- Set MINIMUM_MISSION_VERSION_AR above the current version and try loading that.
- Load a pre-mod community mission.
- Load a Barbolani mission.

### CODE REVIEW

File: A3A/addons/core/functions/Debug/fn_prepFunctions.sqf [hunk 1]
```diff
@@ -126,7 +126,7 @@ A3A_postInitFuncs = [];
 } forEach ("true" configClasses (configFile/"A3A"/"CfgFunctions"));
 
 if (_skipPreInit) exitWith { true };
-if !( isClass (missionConfigFile/"A3A") ) exitWith { false }; //dont run code unless we are in a A3A mission
```
Comment (HakonRydland) [inline A3A/addons/core/functions/Debug/fn_prepFunctions.sqf:129]: note that this will make the function run for any mission, not just antistasi missions

**Review by HakonRydland — APPROVED** [review]
seems like you got all cases covered, and we are bailing still bailing on none antistasi missions, one note for the future is that we need to be cerefull with new preInit/postInit functions added in regards to non antistasi missions being played while the mod is loaded

File: A3A/addons/core/functions/Debug/fn_prepFunctions.sqf (hunk 1 above)
Comment (jaj22) [reply A3A/addons/core/functions/Debug/fn_prepFunctions.sqf:129]: Which they already do with debug disabled. Unless we switch to using the prepFunctions wrapper regardless, it has to be the responsibility of the individual preInit and postInit functions to detect whether they should be doing anything.

**Comment by HakonRydland** [comment]
as a note script version is generated and should never have any other content added to it as it would be overwritten

**Pull request merged**